ratpack-manual/src/content/chapters/37-pac4j.md

```diff
@@ -0,0 +1,94 @@
+# Security using pac4j
+
+The `ratpack-pac4j` extension provides authentication support based on [pac4j](https://github.com/pac4j/pac4j).
+
+`pac4j` is a generic authentication engine which is implemented by many frameworks and supports multiple protocols: OAuth, CAS, OpenID (Connect), SAML, Google App Engine and HTTP (form and basic auth).
+
+The `RatpackPac4j` class is the main component to use to manage the security of your web application.
+
+
+## Prerequisite
+
+All authentication mechanisms work by redirecting the user to an identity provider and finishing the login process when he is redirected back to the application with some dedicated credentials.
+To make these authentication processes work and to save the originally requested url, the session is required. That's why the `SessionModule` is necessary:
+
+```language-java
+.registry(Guice.registry(b -> b
+                  .module(SessionModule.class)
+```
+
+
+## Authentication mechanisms definition
+
+Each authentication mechanism in `pac4j` is defined as a [Client](https://github.com/pac4j/pac4j/blob/master/pac4j-core/src/main/java/org/pac4j/core/client/Client.java).
+Thus, configuring the ways to login into your application requires to create the right [clients](https://github.com/pac4j/pac4j/wiki/Clients): a `FacebookClient` to login with Facebook, a `TwitterClient` to login with Twitter, a `Google2Client` to login with Google (using OAuth v2.0 protocol), and so on.
+
+Create your clients:
+
+```language-java
+final Saml2Client saml2Client = new Saml2Client();
+saml2Client.setKeystorePath("resource:samlKeystore.jks");
+saml2Client.setKeystorePassword("pac4j-demo-passwd");
+saml2Client.setPrivateKeyPassword("pac4j-demo-passwd");
+saml2Client.setIdpMetadataPath("resource:testshib-providers.xml");
+
+final FacebookClient facebookClient = new FacebookClient("145278422258960", "be21409ba8f39b5dae2a7de525484da8")
+final TwitterClient twitterClient = new TwitterClient("CoxUiYwQOSFDReZYdjigBA", "2kAzunH5Btc4gRSaMr7D7MkyoJ5u1VzbOOzE8rBofs");
+
+final FormClient formClient = new FormClient("/theForm.html", new SimpleTestUsernamePasswordAuthenticator(), new UsernameProfileCreator());
+final BasicAuthClient basicAuthClient = new BasicAuthClient(new SimpleTestUsernamePasswordAuthenticator(), new UsernameProfileCreator());
+
+final CasClient casClient = new CasClient();
+casClient.setCasLoginUrl("http://localhost:8888/cas/login");
+```
+
+And define them **BEFORE** any other authentication operation in your web chain using the `RatpackPac4j.authenticator(Client<?, ?>... clients)` method:
+
+```language-java
+chain
+   .all(RatpackPac4j.authenticator(formClient, formClient, saml2Client, facebookClient, twitterClient, basicAuthClient, casClient))
+```
+
+The above command creates the `/authenticator` url endpoint to receive callbacks and finish the login processes (it can be set to an another `path` using `RatpackPac4j.authenticator(String path, Client<?, ?>... clients)`.
+
+
+## Url protection
```

Comment:
The Javadoc is great to know how you should exactly use a method or a class and from this point of view, I think that the Javadoc of the `RatpackPac4j` class is perfect.
But the Javadoc is not very good to give readers an overview and this is what this manual tries to do: what is the prerequisite, how you define the ways to login, how you protect an url, how you get the current user profile, how you logout.

My idea was to shorten the source code to make the user focus on the explanations and you want me to link more to the Javadocs.

Here is my proposal: remove the partial source codes to only keep the explanations and add more links to the Javadoc. Does it sound good to you?